lib/github/commits.rb

```diff
@@ -0,0 +1,58 @@
+# frozen_string_literal: true
+
+module GitHub
+  class Commits < Items
+    def initialize(arr_or_options)
+      super arr_or_options, GitHub::Commit
+    end
+
+    def unique_dco_signers_csv
+      # Create an association list of all name->email pairs, e.g:
```

Comment:
Adopt the `@var ||= begin` syntax to avoid re-parsing/sorting.

- Create a class `Signer` that represents a single DCO signer with name and multiple emails.
- Create a class `Signers` that represents a collection of `Signer` and has a class method that implements the logic here to create it.
- Rename this method to `signers` that returns `Signers`.
- Do sorting in `Signers.sort!`.
- Transform to text elsewhere, e.g. `Signers#to_s`.

This is what this could look like:

```ruby
signers_list.each do |name, email|
  signers[email] ||= Set.new
  signers[email] += email
end
```